spec.html

```diff
@@ -36386,7 +36389,6 @@ <h1>Runtime Semantics: InternalizeJSONProperty ( _holder_, _name_ )</h1>
                   1. Perform ? _val_.[[Delete]](! ToString(_I_)).
                 1. Else,
                   1. Perform ? CreateDataProperty(_val_, ! ToString(_I_), _newElement_).
-                  1. NOTE: This algorithm intentionally does not throw an exception if CreateDataProperty returns *false*.
```

Comment:
This algorithm ignores failures of both `CreateDataProperty` and `[[Delete]]`, but has notes only about ignoring `CreateDataProperty`.

Reply:
Do other places where [[Delete]] return false throw an exception?

Reply:
Yes, two: [`delete` operator](https://tc39.es/ecma262/#sec-delete-operator-runtime-semantics-evaluation) (step 5.b) and [`DeletePropertyOrThrow`](https://tc39.es/ecma262/#sec-deletepropertyorthrow) (step 4).

Handling `false` results of `[[Delete]]` and `[[DefineOwnProperty]]` looks very straightforward in spec with [`DeletePropertyOrThrow`](https://tc39.es/ecma262/#sec-deletepropertyorthrow) and [`DefinePropertyOrThrow`](https://tc39.es/ecma262/#sec-definepropertyorthrow) respectively.

However, some engines (JSC at least) don't have those ops and pass around a "should throw" flag as an argument. A note would be nice, IMO.